packages/core/src/components/graphs/area.ts

```diff
@@ -37,19 +42,70 @@ export class Area extends Component {
 				.x((d, i) => cartesianScales.getDomainValue(d, i))
 				.y0(cartesianScales.getRangeValue(0))
 				.y1((d, i) => cartesianScales.getRangeValue(d, i));
+			domain = this.services.cartesianScales.getMainYScale().domain();
 		} else {
 			areaGenerator
 				.x0(cartesianScales.getRangeValue(0))
 				.x1((d, i) => cartesianScales.getRangeValue(d, i))
 				.y((d, i) => cartesianScales.getDomainValue(d, i));
+			domain = this.services.cartesianScales.getMainXScale().domain();
 		}
 
 		// Update the bound data on area groups
 		const groupedData = this.model.getGroupedData();
+
```

Comment:
i'd just suggest using a variable for `this.model.getOptions()` so you only get the options from the model once in this block